### ISSUE TITLE
nwggrid or nwgbar wont launch

### ISSUE THREAD

**Issue by mybigman** [nwg-piotr/nwg-launchers#162]
WM: sway
Locale: en
terminate called after throwing an instance of 'Gdk::PixbufError'
fish: Job 1, 'nwggrid' terminated by signal SIGABRT (Abort)

fedora 34

**Comment by nwg-piotr** [comment]
Are you trying to run it for the 1st time? Nothing has been changed in the code for months.

**Comment by talzion12** [comment]
I'm seeing this error on NixOS as well. Version 0.4.4.
Any way I can help debug this?

**Comment by nwg-piotr** [comment]
I wish I knew... I'd like to see the crash locally. I'll try to set up Fedora on my testing machine.

For now instead of `nwg-grid` you could try [nwg-drawer](https://github.com/nwg-piotr/nwg-drawer). It does the same plus some extras.

**Comment by talzion12** [comment]
nwg-drawer works great!

nwg-launchers 0.5 still doesn't seem to work for me.

Awesome project btw!

**Comment by Siborgium** [comment]
@talzion12 could you run [this](https://github.com/Siborgium/nwg-launchers/tree/collapse) branch? At least it should give you meaningful error message.

**Comment by talzion12** [comment]
I'm kind of a NixOS newbie so I hope I was able to change the branch correctly in the derivation (I think I did but still).

This is the output I got:
```
➜  nixpkgs git:(nixos-unstable) ✗ nwggrid
INFO: wm: sway
INFO: Locale: en
INFO: Using css file '"/home/tal/.config/nwg-launchers/nwggrid/style.css"'
ERROR: Couldn’t recognize the image file format for file “/nix/store/lbq5w7zjbg2mc26pxh06wczkcg4qh0f2-nwg-launchers-collapse/share/nwg-launchers/nwgbar/icon-missing.svg”
➜  nixpkgs git:(nixos-unstable) ✗ nwgbar
INFO: wm: sway
INFO: 4 bar entries loaded
INFO: Using css file '"/home/tal/.config/nwg-launchers/nwgbar/style.css"'
terminate called after throwing an instance of 'Gdk::PixbufError'
fish: Job 1, 'nwgbar' terminated by signal SIGABRT (Abort)
➜  nixpkgs git:(nixos-unstable) ✗ nwgdmenu
INFO: wm: sway
INFO: Using css file '"/home/tal/.config/nwg-launchers/nwgdmenu/style.css"'
INFO: 1325 commands found

** (nwgdmenu:1414200): WARNING **: 14:16:13.585: gtk-layer-shell v0.6.0 may not work on GTK v3.24.30. If you experience crashes, check https://github.com/wmww/gtk-layer-shell/blob/master/compatibility.md
```

The only one that didn't crash was `nwgdmenu`.

Looks like the important line is this: `ERROR: Couldn’t recognize the image file format for file “/nix/store/lbq5w7zjbg2mc26pxh06wczkcg4qh0f2-nwg-launchers-collapse/share/nwg-launchers/nwgbar/icon-missing.svg”`.

It seems like svg files aren't supported for some reason?

**Comment by Siborgium** [comment]
Yeah, it seems so.  If it's the case, we'll just have to either add it as a hard dependency or provide .jpg or .bmp as fallback of a fallback :^)

Please, post output of `gdk-pixbuf-query-loaders`. If you do not have this utility, check if you have `librsvg` installed. On most distributions it's a hard dependency of gtk, that's why we incorrectly assumed it's always the case. I'll add it to the build file in the next release which is happening in the next few days (I hope :crossed_fingers:). For now you can just install it yourself and run `gdk-pixbuf-query-loaders --update-cache` to be doubly sure.

Hope it helps!

**Comment by talzion12** [comment]
Yes! I installed librsvg and now both nwggrid and nwgbar are working.

@mybigman does it also work for you after installing librsvg?

**Comment by alebastr** [comment]
@Siborgium unfortunately, build dependency does not imply runtime dep in binary distributions, so https://github.com/nwg-piotr/nwg-launchers/commit/c64cee0bf3a1d07cea5bddc8202cab5b1588da59 changes nothing on Fedora. We are generating automatic runtime deps from `ldd` output on all packaged binaries, which does not include required gdk-pixbuf plugins and, AFAIK, Debian packaging guidelines recommend the same.

Your best bet is to ask packagers to explicitly add the dependency in release notes, or make runtime fallback to the png version.
I'll fix Fedora package with 0.6.x update.

**Comment by Siborgium** [comment]
Thanks for the info! I'll update the release description and README.
